matdeeplearn/common/data.py

```diff
@@ -71,39 +74,35 @@ def get_dataset(
     data_path: str
         path to the folder containing data.pt file
 
-    target_index: int
-        the index to select the target values
-        this is needed because in our target.csv, there might be
-        multiple columns of target values available for that
-        particular dataset, thus we need to index one column for
-        the current run/experiment
-
-    transform_type: transformation function/class to be applied
+    transform_list: transformation function/classes to be applied
     """
 
+    transforms = []
     # set transform method
-    if transform_type == "GetY":
-        T = GetY
-    else:
-        raise ValueError("No such transform found for {transform}")
+    for transform in transform_list:
+        if transform["otf"]:
+            transforms.append(
+                registry.get_transform_class(
+                    transform["name"],
+                    **({} if transform["args"] is None else transform["args"])
+                )
+            )
 
     # check if large dataset is needed
     if large_dataset:
         Dataset = LargeStructureDataset
     else:
         Dataset = StructureDataset
 
-    transform = T(index=target_index)
+    composition = Compose(transforms) if len(transforms) > 0 else None
```

Comment:
See the comment on `GetY` above. If we always default to having at least `GetY`, then transforms should always be at least length 1

Reply:
Yes, that makes sense. And instead of defaulting to `None`, we should use `transforms[0]`.